src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -97,93 +134,19 @@ public void BinaryOperatorOverloadResolution(BinaryOperatorKind kind, BoundExpre
-                bool rightHadUserDefinedCandidateFromInterfaces = false;
-                string name = OperatorFacts.BinaryOperatorNameFromOperatorKind(kind);
-                var lookedInInterfaces = PooledDictionary<TypeSymbol, bool>.GetInstance();
-
-                if (!leftHadUserDefinedCandidate)
-                {
-                    leftHadUserDefinedCandidateFromInterfaces = GetUserDefinedBinaryOperatorsFromInterfaces(kind, name,
-                        leftOperatorSourceOpt, leftSourceIsInterface, left, right, ref useSiteDiagnostics, lookedInInterfaces, resultFromInterfaces.Results);
-                }
-
-                if (!rightHadUserDefinedCandidate && leftOperatorSourceOpt != rightOperatorSourceOpt)
-                {
-                    var rightOperators = ArrayBuilder<BinaryOperatorAnalysisResult>.GetInstance();
-                    rightHadUserDefinedCandidateFromInterfaces = GetUserDefinedBinaryOperatorsFromInterfaces(kind, name, 
-                        rightOperatorSourceOpt, rightSourceIsInterface, left, right, ref useSiteDiagnostics, lookedInInterfaces, rightOperators);
-
-                    AddDistinctOperators(resultFromInterfaces.Results, rightOperators);
-                    rightOperators.Free();
-                }
-
-                Debug.Assert(resultFromNonInterfaces != null || 
-                             lookedInInterfaces.Count == 0 ||
-                             (!leftHadUserDefinedCandidate && !rightHadUserDefinedCandidate));
-
-                if (leftHadUserDefinedCandidateFromInterfaces || rightHadUserDefinedCandidateFromInterfaces)
-                {
-                    result.Clear();
-
-                    if (resultFromNonInterfaces != null)
-                    {
-                        result.Results.AddRange(resultFromNonInterfaces.Results);
-                    }
-
-                    result.Results.AddRange(resultFromInterfaces.Results);
-                    BinaryOperatorOverloadResolution(left, right, result, ref useSiteDiagnostics);
-                }
-
-                resultFromInterfaces.Free();
-                lookedInInterfaces.Free();
-            }
-
-            if (resultFromNonInterfaces != null)
-            {
-                resultFromNonInterfaces.Free();
-            }
-        }
-
-        private static bool ShouldGetUserDefinedBinaryOperatorsFromInterfaces(
-            TypeSymbol operatorSourceOpt, bool sourceIsInterface,
-            ref HashSet<DiagnosticInfo> useSiteDiagnostics)
-        {
-            return (object)operatorSourceOpt != null &&
-                   (sourceIsInterface ||
-                    (operatorSourceOpt.IsTypeParameter() &&
-                     !((TypeParameterSymbol)operatorSourceOpt).AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics).IsEmpty));
         }
 
         private bool GetUserDefinedBinaryOperatorsFromInterfaces(BinaryOperatorKind kind, string name, 
```

Comment:
>GetUserDefinedBinaryOperatorsFromInterfaces [](start = 21, length = 43)

This implementation does not implement the shadowing rules from the LDM. In [https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-06-27.md](https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-06-27.md) it says

> If we find an **applicable candidate** in an interface, that candidate shadows all applicable operators in base interfaces: we stop looking. This is a bit more complicated than in classes, where it's linear, but we can express it right.

My emphasis. Whether a declared operator is **applicable** to the arguments is computed later, during overload resolution. Therefore, the shadowing has to be computed later, not here in name lookup.